File: app/src/main/java/com/example/david/lists/view/itemlist/ItemDiffCallback.java
package com.example.david.lists.view.itemlist;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import com.example.david.lists.data.datamodel.Item;

class ItemDiffCallback extends DiffUtil.ItemCallback<Item> {
    @Override
    public boolean areItemsTheSame(@NonNull Item oldItem, @NonNull Item newItem) {
        return oldItem.getId().equals(newItem.getId());
    }

    @Override
    public boolean areContentsTheSame(@NonNull Item oldItem, @NonNull Item newItem) {
        return oldItem.toString().equals(newItem.toString());
    }
}

File: app/src/main/java/com/example/david/lists/view/itemlist/ItemDiffCallback.kt
package com.example.david.lists.view.itemlist

import androidx.recyclerview.widget.DiffUtil

import com.example.domain.datamodel.Item

class ItemDiffCallback : DiffUtil.ItemCallback<Item>() {
    override fun areItemsTheSame(oldItem: Item, newItem: Item) =
            oldItem.id == newItem.id

    override fun areContentsTheSame(oldItem: Item, newItem: Item) =
            oldItem == newItem
}
